### PR TITLE
Split gemfiles for testing against multiple versions of Rails

### PR DESCRIPTION
**Pull request by jesjos**
Rationale: Different Gemfiles -> Different Gemfile.lock files.

This makes switching between rails versions much cheaper.

### CODE REVIEW

**Comment by jesjos** [comment]
Hey @benedikt would love your input on this. There's a bit of overhead in reading multiple Gemfiles. But it's much easier to switch between Gemfiles when developing.

**Comment by benedikt** [comment]
@jesjos Nice approach!

Two thoughts: 

1. Instead of setting the `rails_version` in each Gemfile, there could just as well be the `gem 'rails'` statements with the version requirements in there. This would remove the need for the additional check in `Gemfile.base`. 

2. The lock files should probably not be included in the repository (like `Gemfile.lock` they should be in `.gitignore`).

**Comment by jesjos** [comment]
@benedikt expert suggestions, I acted accordingly! Also: used `eval_gemfile`

File: Gemfile.rails-master [hunk 1]
```diff
@@ -0,0 +1,5 @@
+group :development, :test do
+  gem 'rails' github: 'rails/rails'
```
Comment (benedikt) [inline Gemfile.rails-master]: Missing comma 🙂

**Comment by benedikt** [comment]
Nice! Didn’t know about `eval_gemfile`.

**Pull request merged**